chromium_src/net/tools/transport_security_state_generator/input_file_parsers.cc

```diff
@@ -0,0 +1,326 @@
+BggrBgEFBQcwAYYmaHR0cDovL2lzcmcudHJ1c3RpZC5vY3NwLmlkZW50cnVzdC5j
+b20wOwYIKwYBBQUHMAKGL2h0dHA6Ly9hcHBzLmlkZW50cnVzdC5jb20vcm9vdHMv
+ZHN0cm9vdGNheDMucDdjMB8GA1UdIwQYMBaAFMSnsaR7LHH62+FLkHX/xBVghYkQ
+MFQGA1UdIARNMEswCAYGZ4EMAQIBMD8GCysGAQQBgt8TAQEBMDAwLgYIKwYBBQUH
+AgEWImh0dHA6Ly9jcHMucm9vdC14MS5sZXRzZW5jcnlwdC5vcmcwPAYDVR0fBDUw
+MzAxoC+gLYYraHR0cDovL2NybC5pZGVudHJ1c3QuY29tL0RTVFJPT1RDQVgzQ1JM
+LmNybDATBgNVHR4EDDAKoQgwBoIELm1pbDAdBgNVHQ4EFgQUxbGrTkyxzWQwk37B
+hJkFq+YD4iUwDQYJKoZIhvcNAQELBQADggEBAAcSAhaE7rvHxyUnhgkEpMR56o2I
+IH+mlw5kknjhAuvaBIAM59MZkFbFg5CrNWt8K+G3UoxJgFwv7HvJJxqwgPpNgXC/
+uT3prkvwt+2lvzKJKbqdH+lo40P8EuSyyJOz2hjrRzNMHbJHYDS9OhF5WC5LOQQa
+ydgLZ/JHxXgJypEZqcmVgQ+yYBs0XPwXjE7OE8vbx5REwu7gToMIqAoWRoWW2MxS
+g28RGPVnHzHk2XV1nZGy9T+NYQ91vWWJr1pzNEFZ0cnA2xGwTeJ+zZ3URCfw3Z1U
++YAL3YUmrvdoRBlASOTmNJmXSo9qvMYPa3DEomAPoFQFZqsSN6kuqDEIqMA=
+-----END CERTIFICATE-----)brave_certs";
+
+  return ParseCertificatesFile_ChromiumImpl(brave_certs, pinsets);
+}
+
+bool ParseJSON(base::StringPiece json,
+               TransportSecurityStateEntries* entries,
+               Pinsets* pinsets) {
+  Pinsets chromium_pinsets;
+  TransportSecurityStateEntries chromium_entries;
+  if (!ParseJSON_ChromiumImpl(json, &chromium_entries, &chromium_pinsets))
+    return false;
+
+  for (auto& entry : chromium_entries) {
+    // leave test entries
+    if (entry->pinset != "test")
+      entry->pinset = "";
+
+    if (!entry->force_https && entry->pinset.empty() && !entry->expect_ct)
+      continue;
+
+    entries->push_back(std::move(entry));
+  }
+
+  base::StringPiece brave_json = R"brave_json({
+    "pinsets": [
+      {
+        "name": "test",
+        "static_spki_hashes": [
+          "TestSPKI"
+        ],
+        "report_uri": "http://report-example.test/test"
+      },
+      {
+        "name": "brave",
+        "static_spki_hashes": [
+          "AmazonRootCA1",
+          "AmazonRootCA2",
+          "AmazonRootCA3",
+          "AmazonRootCA4",
+          "StarfieldG2",
+          "GlobalSignRootCA",
+          "GlobalSignRootCA_R2",
+          "GlobalSignRootCA_R3",
+          "LetsEncryptAuthorityPrimary_X1_X3",
+          "LetsEncryptAuthorityBackup_X2_X4"
+        ]
+      }
+    ],
+    "entries": [
+      // Brave
+      { "name": "ledger.mercury.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "ledger-staging.mercury.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "balance.mercury.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "balance-staging.mercury.basicattentiontoken.org", "policy": "custom", "mode": "force-https", "pins": "brave"},
+      { "name": "download.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
+      { "name": "laptop-updates.brave.com", "mode": "force-https", "policy": "custom", "pins": "brave"},
```

Comment:
does anyone know yet if we're going to be using the same domains for brave-core updates? if not we need to add a new issue to add them to this list once they're finalized

Reply:
cc @bbondy @aekeus

Reply:
cc @RyanJarv